deps/ledger-zxlib/include/zxformat.h

```diff
@@ -96,15 +97,86 @@ __Z_INLINE void bip32_to_str(char *s, uint32_t max, const uint32_t *path, uint8_
             written = strlen(s + offset);
             if (written == 0 || written >= max - offset) {
                 snprintf(s, max, "ERROR");
-                return;
+                return 0;
             }
             offset += written;
         }
     }
+
+    return offset;
+}
+
+__Z_INLINE uint32_t bip44_to_str(char *s, uint32_t max, const uint32_t path[5]) {
+    return bip32_to_str(s, max, path, 5);
+};
+
+#define SECP256R1_STRING " SECP256R1"
+#define SECP256K1_STRING " SECP256K1"
+#define SHA2_256_STRING " SHA-2"
+#define SHA3_256_STRING " SHA-3"
+#define DESIRED_HD_PATH_LENGTH 17
+
+__Z_INLINE uint32_t add_options_to_path(char *s, uint32_t max, uint16_t options) {
+    uint32_t written = strlen(s);
+    uint8_t curve = (options >> 8);
+    uint8_t hash = options & 0xFF;
+
+    //For better UI if path is short, we add one or two spaces
+    if (written < DESIRED_HD_PATH_LENGTH && max >= DESIRED_HD_PATH_LENGTH+1) {
+        for(;written <= DESIRED_HD_PATH_LENGTH; written++) {
+            s[written] = ' ';
+        }
+        s[written] = 0;
+        written--;
+    }
+
+    if (curve != 0) {
+        if (written + sizeof(SECP256R1_STRING) > max || written + sizeof(SECP256K1_STRING) > max) {
+            snprintf(s, max, "ERROR");
+            return 0;
+        }
+        switch(curve) {
+            case 0x02:
+                snprintf(s+written, max, SECP256R1_STRING);
+                written += sizeof(SECP256R1_STRING) - 1;
+                break;
+            case 0x03:
+                snprintf(s+written, max, SECP256K1_STRING);
+                written += sizeof(SECP256R1_STRING) - 1;
```

Comment:
Should this be "SECP256K1" for correctness? I know the actual strings are the same size.